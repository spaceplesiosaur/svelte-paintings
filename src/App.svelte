<script>
	export let name;
	import { onMount } from 'svelte';
	const apiKey = "29c0f9d0-5b77-11ea-87eb-f9c17db1ada1"
	const fetchUrl = `https://api.harvardartmuseums.org/object?person=35055&apikey=${apiKey}`
	const artwork = []
	const photos = []
	const paintings = []
	const sketches = []
	const bronze = []
	const misc = []
	let count = 0
	let currentMedium = misc

	const getArt = async (endpoint) => {
		if (endpoint) {
				let res = await fetch(endpoint);
				let jsonRes = await res.json();
				artwork.push(...jsonRes.records)
				getArt(jsonRes.info.next)
			} else {
				sortArt()
			}
	}

	const sortArt = () => {
		artwork.forEach(piece => {
			switch (piece.classification) {
	      case "Photographs":
	        photos.push({title: piece.title, image: piece.images, colors: piece.colors, date: piece.dateend});
					photos = photos
					break
	      case "Drawings":
	        sketches.push({title: piece.title, image: piece.images, colors: piece.colors, date: piece.dateend});
					sketches = sketches
					break
	      case "Paintings":
	        paintings.push({title: piece.title, image: piece.images, colors: piece.colors, date: piece.dateend});
					paintings = paintings
					break
	      case "Sculpture":
	        bronze.push({title: piece.title, image: piece.images, colors: piece.colors, date: piece.dateend});
					bronze = bronze
					break
	      default:
	        misc.push({title: piece.title, image: piece.images, colors: piece.colors, date: piece.dateend});
					misc = misc
    	}
		})
	}

	const chooseCurrentMedium = (medium) => {
		currentMedium = medium
		console.log("current", currentMedium)
	}

	const dealWithImages = (image) => {
		if (image === undefined) {
			return null
		} else {
			return image[0].baseimageurl
		}
	}

	onMount(async () => {
		getArt(fetchUrl)
	});

</script>

<main>
	<h1>Ponies and Ballerinas!</h1>
	<p class="subheader">Edgar Degas is like an eight-year-old girl.  It's fantastic.</p>
	<nav>
		<button class="nav-button" id="photos" on:click="{chooseCurrentMedium(photos)}">Photographs</button>
		<button class="nav-button" id="sketches" on:click={chooseCurrentMedium(sketches)}>Drawings</button>
		<button class="nav-button" id="paintings" on:click={chooseCurrentMedium(paintings)}>Paintings</button>
		<button class="nav-button" id="bronze" on:click={chooseCurrentMedium(bronze)}>Sculptures</button>
		<button class="nav-button" id="misc" on:click={chooseCurrentMedium(misc)}>Other</button>
	</nav>
	<h3 class="instructions">Click a button to see Degas's artwork by medium, then journey through the gallery...</h3>
	<div class="art-container">
		{#each currentMedium as item}
			<div class="art-card">
				<img class="card-image" src="{dealWithImages(item.image)}">
				<h2 class="card-title">{item.title}</h2>
				<p class="card-date">{item.date}</p>
			</div>
		{/each}
	</div>
</main>

<style>

	main {
		text-align: center;
		padding: 1em;
		max-width: 240px;
		margin: 0 auto;
		background-color: #FAF7F4;
		min-height: 800px;
	}

	h1 {
		color: #2b2d2f;
		text-transform: uppercase;
		font-size: 4em;
		font-weight: 100;
		background-color: #f6e1e6;
	}

	.subheader {
		font-size: 2em;
	}

	.nav-button {
		margin: 1em;
		min-width: 200px;
		height: 75px;
		background-color: #f6e1e6;
		color: #2b2d2f;
		font-size: 1.5em;
		border: 2px solid #8AB9D3
	}

	.art-container {
		display: flex;
		flex-flow: column wrap;
		background-color: #8AB9D3;
		justify-content: center;
		max-height: 1000px;
  	margin-left: -8px;
		overflow-y: scroll;
		overflow-x: inherit;
	}
	.art-card {
  	margin: 10px;
		border: 2px solid black;
		background-color: #FAF7F4;
	}
	.instructions {
		align-items: center;
	}
	.card-image {
		max-width: 250px;
		overflow: scroll;
		margin: 5px;
		border: 2px solid black;
	}
	.card-title {
		max-width: 250px;
		font-size: 1.5em;
		align-items: center;
		margin: 1px;
	}

	.card-date {
		max-height: 100px;
		font-size: 1em;
		align-items: center;
	}

	@media (min-width: 640px) {
		main {
			max-width: none;
		}
	}
</style>
