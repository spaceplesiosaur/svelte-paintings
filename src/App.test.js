import '@testing-library/jest-dom/extend-expect'

import { render, fireEvent } from '@testing-library/svelte'

import App from './App'


test('shows app name in header when rendered', () => {
  const { getByText } = render(App, { name: 'Inferno' })

  expect(getByText('Ponies and Ballerinas!')).toBeInTheDocument()
})
