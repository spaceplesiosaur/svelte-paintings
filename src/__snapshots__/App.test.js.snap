// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
Object {
  "component": App {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [
        Array [],
        Array [],
        Array [],
        Array [],
        Array [],
        Array [],
        [Function],
        [Function],
        "Inferno",
      ],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
      ],
      "on_mount": Array [],
      "props": Object {
        "name": 8,
      },
      "update": [Function],
    },
    "$capture_state": [Function],
    "$inject_state": [Function],
    "$set": [Function],
  },
  "container": <body>
    <div>
      <main
        class="svelte-1t76zzn"
      >
        <h1
          class="svelte-1t76zzn"
        >
          Ponies and Ballerinas!
        </h1>
         
        <p
          class="subheader svelte-1t76zzn"
        >
          Edgar Degas is like an eight-year-old girl.  It's fantastic.
        </p>
         
        <nav>
          <button
            class="nav-button svelte-1t76zzn"
            id="photos"
          >
            Photographs
          </button>
           
          <button
            class="nav-button svelte-1t76zzn"
            id="sketches"
          >
            Drawings
          </button>
           
          <button
            class="nav-button svelte-1t76zzn"
            id="paintings"
          >
            Paintings
          </button>
           
          <button
            class="nav-button svelte-1t76zzn"
            id="bronze"
          >
            Sculptures
          </button>
           
          <button
            class="nav-button svelte-1t76zzn"
            id="misc"
          >
            Other
          </button>
        </nav>
         
        <h3
          class="instructions svelte-1t76zzn"
        >
          Click a button to see Degas's artwork by medium, then journey through the gallery...
        </h3>
         
        <div
          class="art-container svelte-1t76zzn"
        />
      </main>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
